server/src/main/java/com/linecorp/centraldogma/server/internal/api/ContentServiceV1.java

```diff
@@ -406,12 +407,24 @@ private static Object handleWatchFailure(Throwable thrown) {
     }
 
     /**
-     * Checks if the commit is for mirroring setting and raises an exception if the {@code localRepo} field
-     * is one of {@code meta} and {@code dogma} which are internal repositories.
+     * Checks if the commit is for creating a file and raises a {@link InvalidPushException} if the
+     * given {@code repoName} field is one of {@code meta} and {@code dogma} which are internal repositories.
      */
-    public static void checkMirrorLocalRepo(String repoName, Iterable<Change<?>> changes) {
-        // TODO(minwoox): Provide an internal API for mirroring setup with a better UI(?) and check this there.
+    public static void checkPush(String repoName, Iterable<Change<?>> changes) {
+        if (Project.REPO_DOGMA.equals(repoName)) {
+            throw new InvalidPushException(
+                    "The " + Project.REPO_DOGMA + " repository is reserved for internal usage.");
+        }
+
         if (Project.REPO_META.equals(repoName)) {
+            final boolean hasChangesWithoutMirroring =
+                    Streams.stream(changes)
+                           .anyMatch(change -> !DefaultMetaRepository.PATH_MIRRORS.equals(change.path()));
```

Comment:
Could we check the permission of the current user? We should allow only `OWNER` permission.
https://github.com/line/centraldogma/blob/082d87bd172fc0705579f2774458466e0dbfc1c6/server/src/main/java/com/linecorp/centraldogma/server/internal/api/MetadataApiService.java#L63

Reply:
Hmm, good suggestion. Let me find a way.

Reply:
How about calling `findRole` while checking and returning `CompletableFutre<Void>`?

https://github.com/line/centraldogma/blob/81d0be9597ad409ebbae90f9884eb85e129931a9/server/src/main/java/com/linecorp/centraldogma/server/metadata/MetadataService.java#L786

Reply:
I think we don't have to because `RequiresPermissionDecorator` handles it.
https://github.com/line/centraldogma/blob/master/server/src/main/java/com/linecorp/centraldogma/server/internal/api/auth/RequiresPermissionDecorator.java#L90

Reply:
Ah, you are right.